src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+     */
+    private static long getLeapYearsBefore(final long prolepticYear) {
+        long yearBefore = prolepticYear - 1;
+
+        return (yearBefore / 4) - (yearBefore / 100) + (yearBefore / 400);
+    }
+
+    /**
+     * Factory method, validates the given triplet year, month and dayOfMonth
+     * Leap day or year day must be created with {@link InternationalFixedDate#create(int,boolean,boolean)}.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from 1 to 13
+     * @param dayOfMonth    the International fixed day-of-month, from 1 to 28
+     * @return the International fixed date
+     * @throws DateTimeException if the date is invalid
+     */
+    static InternationalFixedDate create(final int prolepticYear, final int month, final int dayOfMonth) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+        InternationalFixedChronology.MONTH_OF_YEAR_RANGE.checkValidValue(month, ChronoField.MONTH_OF_YEAR);
+        InternationalFixedChronology.DAY_OF_MONTH_RANGE.checkValidValue(dayOfMonth, ChronoField.DAY_OF_MONTH);
+
+        if (1 > prolepticYear) {
+            throw new DateTimeException("Invalid date: " + prolepticYear + '-' + month + '-' + dayOfMonth);
+        }
+
+        return new InternationalFixedDate(prolepticYear, month, dayOfMonth);
+    }
+
+    /**
+     * Factory method, validates the given triplet year, month and dayOfMonth
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param isLeapDay     is it leap-day, month-less day between end of June and beginning of Sol
+     * @param isYearDay     is it year-day, month-less day between end of December of the old year and beginning of January of the following year
+     * @return the International fixed date
+     * @throws DateTimeException if the date is invalid
+     */
+    static InternationalFixedDate create(final int prolepticYear, final boolean isLeapDay, final boolean isYearDay) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+
+        if (1 > prolepticYear) {
+            throw new DateTimeException("Invalid date: " + prolepticYear);
+        }
+
+        if (isLeapDay == isYearDay) {
+            throw new DateTimeException("Invalid date: " + prolepticYear + " both leap-day and year-day are " + isYearDay);
+        }
+
+        if (isLeapDay && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            throw new DateTimeException("Invalid leap day for year: " + prolepticYear);
+        }
+
+        return new InternationalFixedDate(prolepticYear, isLeapDay, isYearDay);
+    }
+
+    /**
+     *
+     * Validates the object.
+     *
+     * @return the resolved date, not null
+     */
+    private Object readResolve() {
+        if (isLeapDay() || isYearDay()) {
```

Comment:
I think I'd recommend ducting directly through `create(...)`, although that's somewhat more optional.